feature(renderer): Support reversed lists

gdamore opened the pull request:
While testing, found some more potential edge cases around
the parser for attributes lists, so tightened those up.

Fixes #662

xcoulon commented:
@gdamore looks good but there is a couple of formatting issues reported by the linter

gdamore commented:
was extra lines (whitespace) in the test file.  (My editor stopped formatting on save for some reason...)

Fixed and squashed.  Should be ready to go.

xcoulon commented:
> was extra lines (whitespace) in the test file. (My editor stopped formatting on save for some reason...)
> 
> Fixed and squashed. Should be ready to go.

thanks!
